Problem-Service/src/main/java/tw/waterball/judgegirl/problemservice/domain/usecases/PatchProblemUseCase.java

```diff
@@ -0,0 +1,58 @@
+package tw.waterball.judgegirl.problemservice.domain.usecases;
+
+import lombok.Value;
+import tw.waterball.judgegirl.commons.exceptions.NotFoundException;
+import tw.waterball.judgegirl.entities.problem.JudgePluginTag;
+import tw.waterball.judgegirl.problemservice.domain.repositories.PatchProblemParams;
+import tw.waterball.judgegirl.problemservice.domain.repositories.ProblemRepository;
+
+import javax.inject.Named;
+import java.util.Set;
+
+import static tw.waterball.judgegirl.commons.exceptions.NotFoundException.notFound;
+
+@Named
+public class PatchProblemUseCase extends BaseProblemUseCase {
+    public PatchProblemUseCase(ProblemRepository problemRepository) {
+        super(problemRepository);
+    }
+
+    public void execute(Request request) throws NotFoundException {
+        if (problemRepository.problemExists(request.problemId)) {
+            problemRepository.patchProblem(
+                    request.problemId,
+                    PatchProblemParams.builder()
+                            .title(request.title)
+                            .description(request.description)
+                            .matchPolicyPluginTag(request.judgePluginTag)
+                            .filterPluginTags(request.filterPluginTags).build());
+        } else {
+            throw notFound("problem").id(request.problemId);
+        }
+    }
+
+    @Value
+    public static class Request {
+        public int problemId;
```

Comment:
default the `problemId` to `-1`
so that if the problemId is not present, it will throw `NotFoundException`